pkg/controller/client_builder.go

```diff
@@ -196,24 +199,27 @@ func (b SAControllerClientBuilder) getAuthenticatedConfig(sa *v1.ServiceAccount,
 			return nil, false, nil
 		}
 		klog.V(4).Infof("Verified credential for %s/%s", sa.Namespace, sa.Name)
-		return clientConfig, true, nil
+		return clientConfigs, true, nil
 	}
 
 	// If we couldn't run the token review, the API might be disabled or we might not have permission.
 	// Try to make a request to /apis with the token. If we get a 401 we should consider the token invalid.
-	clientConfigCopy := *clientConfig
-	clientConfigCopy.NegotiatedSerializer = legacyscheme.Codecs
-	client, err := restclient.UnversionedRESTClientFor(&clientConfigCopy)
-	if err != nil {
-		return nil, false, err
-	}
-	err = client.Get().AbsPath("/apis").Do().Error()
-	if apierrors.IsUnauthorized(err) {
-		klog.Warningf("Token for %s/%s did not authenticate correctly: %v", sa.Namespace, sa.Name, err)
-		return nil, false, nil
+	clientConfigCopy := *clientConfigs
+
+	for _, config := range clientConfigCopy.GetAllConfigs() {
+		config.NegotiatedSerializer = legacyscheme.Codecs
+		client, err := restclient.UnversionedRESTClientFor(config)
+		if err != nil {
+			return nil, false, err
+		}
+		err = client.Get().AbsPath("/apis").Do().Error()
+		if apierrors.IsUnauthorized(err) {
+			klog.Warningf("Token for %s/%s did not authenticate correctly: %v", sa.Namespace, sa.Name, err)
```

Comment:
We might add config host to the warning so that we know which client causes the unauthorized error